### PR TITLE
Move perforce settings to fe

### PR DESCRIPTION
**Pull request by amits1995**

### CODE REVIEW

File: graphql/schema.graphql [hunk 1]
```diff
@@ -65,6 +72,7 @@ input SettingsInput {
   PerforceConnectionString: String
   PerforceUser: String
   PerforceTimeout: String
+  OverrideGlobal: Boolean
```
Comment (dan505512) [inline graphql/schema.graphql:75]: Don't we want to only override some? let's say someone only wants to change the timeout?

File: src/api.ts [hunk 2]
```diff
@@ -17,11 +17,12 @@ import {
   TMP_DIR_PREFIX,
   canAuthGitRepo
 } from "./git";
-import {getPerforceManagerSingleton, IPerforceRepo, IPerforceView} from "./perforceManager";
+import {getPerforceManagerSingleton, IPerforceRepo, IPerforceView, changePerforceManagerSingleton} from "./perforceManager";
 import {Repo, repStore} from "./repoStore";
 import {loadingStateUpdateHandler, onAddRepoRequestHandler} from "./server";
 import {getLogger} from "./logger";
 import { setSettings, getSettings } from "./utils";
+import { isNumber } from "lodash";
```
Comment (dan505512) [inline src/api.ts:25]: When I tried to use isNumber it wasn't that good. I did !isNaN in the end

File: src/api.ts [hunk 3]
```diff
@@ -230,6 +231,20 @@ export const resolvers = {
...
+        const isSuccess = changePerforceManagerSingleton({
+          connectionString: args.connectionSettings.PerforceConnectionString,
+          timeout: parseInt(args.connectionSettings.PerforceTimeout || "5000", 10),
+          username: args.connectionSettings.PerforceUser
+        });
+        return { isSuccess, reason: 'make sure your configuration is correct' }
+      } catch (e) {
+        getLogger("Perforce").error("Failed to connect to Perforce", { e, settings: args.connectionSettings });
+        console.error(`Failed to init perforce manager with port: ${args.connectionSettings}`);
+        return { isSuccess: false, reason: e?.toString() };
```
Comment (dan505512) [inline src/api.ts:245]: Maybe add a fallback reason?

File: graphql/schema.graphql (hunk 1 above)
Comment (dan505512) [reply graphql/schema.graphql:75]: I read the rest and got what you were doing, but how do we know here what's overridden and what's global?

Comment (amits1995) [reply graphql/schema.graphql:75]: The fields that are given are overriden, if for example `PerforceTimeout` wasn't supplied in the request, it wont be overriden

File: src/api.ts (hunk 2 above)
Comment (amits1995) [reply src/api.ts]: not in use - deleting this import

File: src/api.ts (hunk 3 above)
Comment (amits1995) [reply src/api.ts:244]: WDYM?

File: graphql/schema.graphql (hunk 1 above)
Comment (dan505512) [reply graphql/schema.graphql:75]: It's a bit vague as this is a behavior expected from the FE and that might be broken one day by an unsuspecting noob. Comment? ❤️

File: src/api.ts (hunk 3 above)
Comment (dan505512) [reply src/api.ts:244]: If e is undefined we will get no reason for the error, so maybe there should be a default reason there

File: graphql/schema.graphql (hunk 1 above)
Comment (amits1995) [reply graphql/schema.graphql:75]: Let's discuss offline :)

Comment (dan505512) [reply graphql/schema.graphql:79]: We just want to make sure the next dev understands that not giving a param means that this one is coming from the org

**Pull request merged**